frame/atomic-swap/src/lib.rs

```diff
@@ -74,69 +74,63 @@ pub type HashedProof = [u8; 32];
 /// succeeds with best efforts.
 /// - **Claim**: claim any resources reserved in the first phrase.
 /// - **Cancel**: cancel any resources reserved in the first phrase.
-pub trait SwapAction<T: Trait> {
+pub trait SwapAction<AccountId, T: Trait> {
```

Comment:
Is `T` still necessary here? I think it's not referenced at all in all trait methods, right?

Reply:
It is required for the implementation currently to calculate the weight. In the future, when we have reworked the weight calculation, we can change this again.

Reply:
You mean this line, right? https://github.com/paritytech/substrate/pull/6580/files#diff-a87e85c9652bc9cda965e05fa1a82320R130

Do you think it works if we put `T` as a type parameter for `BalanceSwapAction` instead?